src/column/reader.rs

```diff
@@ -694,7 +787,69 @@ mod tests {
         "def content doesn't match");
       assert_eq!(
         &actual_values[..curr_values_read], &self.values[..],
-        "values doesn't match");
+        "values content doesn't match");
+    }
+
+    // Helper function to test read_batch with custom buffers for values, definition and
+    // repetition levels.
+    fn test_read_batch(
```

Comment:
there seem to be some duplicated code in this and `test_plain`, but I think we can address that later in a separate PR.

Reply:
Yes, I will create an issue to update tests `test_plain`, `test_dict` and `test_read_batch` into some common method(s) or structures.